react/components/UI/Modal/index.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+import { provide } from 'react/apollo';
+import mount from 'react/util/mount';
+import unmount from 'react/util/unmount';
+
+import ModalComponent from 'react/components/UI/Modal/Modal';
+
+export default class Modal {
+  ModalComponent = ModalComponent
+
+  constructor(Component, props = {}) {
+    this.Component = Component;
+    this.props = props;
+    this.el = document.createElement('div');
+  }
+
+  open = () => {
+    document.body.appendChild(this.el);
+
+    this.Provided = provide(this.Component, {
+      onClose: this.close,
+      ...this.props,
+    });
```

Comment:
This is slightly weird. I might re-implement this using portals. Or have it be an option. The modal winds up being in a separate React DOM tree. I figured we might want to build multiple UIs as modals that are distinct while migrating over. Since this is client-side they wind up sharing the same Apollo store anyway.

Reply:
Hm yeah. I'll add that it took a second for me to parse `provide` as terminology (env though I love the idea). Maybe something more descriptive like `wrapApollo` is more quickly understandable.